### ISSUE TITLE
Building without `extension-module` warning when only set in `pyproject.toml`

### ISSUE THREAD

**Issue by acshi** [PyO3/maturin#1029]
### Bug Description

When I enable the extension module like so:
```toml
[tool.maturin]
features = ["pyo3/extension-module"]
```

I still get a warning:
```
⚠️  Warning: You're building a library without activating pyo3's `extension-module` feature. See https://pyo3.rs/v0.15.2/building_and_distribution.html#linking
🔗 Found pyo3 bindings
🐍 Found CPython 3.9 at /Users/acshi/gaia/lidar_bitpack/.venv/bin/python
📡 Using build options profile, features from pyproject.toml
```

And I note that maturin only knows that the feature is enabled after is parses `pyproject.toml` but that clearly happens only after the warning has already been emitted.

Using `otool -L` on macos (`ldd` equivalent) confirms that the system python3 has not been linked.

### Your Python version (`python -V`)

Python 3.9.13

### Your pip version (`pip -V`)

pip 22.1.1

### What bindings you're using

pyo3

### Does `cargo build` work?

- [X] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [X] Yes

### Steps to Reproduce

1. Create a new project with `maturin new`.
2. Use `pyo3 = "0.16.5"` in `Cargo.toml` to not directly specify the extension-module feature
3. Set it in `pyproject.toml` with:
```toml
[tool.maturin]
features = ["pyo3/extension-module"]
```
4. `maturin build` gives the warning but properly reads `pyproject.toml` and uses the extension-module feature.

**Comment by messense** [comment]
I just give it a try and did not reproduce the issue.

```bash
$ maturin --version
maturin 0.13.1
$ maturin new hello
    Finished dev [unoptimized + debuginfo] target(s) in 0.18s
     Running `target/debug/maturin new hello`
✔ 🤷 Which kind of bindings to use? · pyo3
  ✨ Done! New project created hello
$ cd hello
$ maturin build
🔗 Found pyo3 bindings
🐍 Found CPython 3.9 at /Users/messense/.pyenv/versions/3.9.11/bin/python3
📡 Using build options features from pyproject.toml
   Compiling target-lexicon v0.12.4
   Compiling once_cell v1.13.0
   Compiling proc-macro2 v1.0.42
   Compiling quote v1.0.20
   Compiling unicode-ident v1.0.2
   Compiling libc v0.2.126
   Compiling syn v1.0.98
   Compiling autocfg v1.1.0
   Compiling parking_lot_core v0.9.3
   Compiling smallvec v1.9.0
   Compiling scopeguard v1.1.0
   Compiling cfg-if v1.0.0
   Compiling indoc v1.0.6
   Compiling unindent v0.1.9
   Compiling lock_api v0.4.7
   Compiling pyo3-build-config v0.16.5
   Compiling parking_lot v0.12.1
   Compiling pyo3-macros-backend v0.16.5
   Compiling pyo3-ffi v0.16.5
   Compiling pyo3 v0.16.5
   Compiling pyo3-macros v0.16.5
   Compiling hello v0.1.0 (/Users/messense/Projects/maturin/hello)
    Finished dev [unoptimized + debuginfo] target(s) in 7.42s
📦 Built wheel for CPython 3.9 to /Users/messense/Projects/maturin/hello/target/wheels/hello-0.1.0-cp39-cp39-macosx_11_0_arm64.whl
$ cat Cargo.toml
[package]
name = "hello"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
[lib]
name = "hello"
crate-type = ["cdylib"]

[dependencies]
pyo3 = "0.16.5"

$ cat pyproject.toml
[build-system]
requires = ["maturin>=0.13,<0.14"]
build-backend = "maturin"

[project]
name = "hello"
requires-python = ">=3.7"
classifiers = [
    "Programming Language :: Rust",
    "Programming Language :: Python :: Implementation :: CPython",
    "Programming Language :: Python :: Implementation :: PyPy",
]

[tool.maturin]
features = ["pyo3/extension-module"]
```

**Comment by messense** [comment]
What version of maturin were you using? I think it's fixed in https://github.com/PyO3/maturin/pull/1014 which is released in v0.13.1.

**Comment by acshi** [comment]
Ahhh, okay, I'm sure it is fixed then. I had the latest from pip which was still 0.13.0.
